src/uu/ls/src/ls.rs

```diff
@@ -1880,13 +1881,27 @@ fn display_item_long(
 
         let dfn = display_file_name(item, config, None, out).contents;
 
-        let _ = writeln!(
-            out,
-            " {} {} {}",
-            pad_left(&display_size_or_rdev(md, config), padding.longest_size_len),
-            display_date(md, config),
-            dfn,
-        );
+        match display_size_or_rdev(md, config) {
+            SizeOrDeviceId::Size(size) => {
+                let _ = writeln!(
+                    out,
+                    " {} {} {}",
+                    pad_left(&size, padding.longest_size_len),
+                    display_date(md, config),
```

Comment:
Instead of duplicating the logic for date and dfn, I think it makes sense to write those separately after the size/rdev.

Reply:
Sure.  Makes sense.